From 0e23b8757182a41e88e8b6692c86bc20096284be Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jo=C3=A3o=20Henrique=20Ferreira=20de=20Freitas?=
 <joaohf@gmail.com>
Date: Sat, 19 Apr 2025 10:17:06 -0300
Subject: [PATCH] erts: fix incompatible pointer type

Expected ErlNifPort instead ErlNifPid:

  nif_SUITE.c:2866:43: error: passing argument 3 of 'enif_whereis_port' from incompatible pointer type [-Wincompatible-pointer-types]

Upstream-Status: Submitted [https://github.com/erlang/otp/pull/9742]
---
 erts/emulator/test/nif_SUITE_data/nif_SUITE.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/erts/emulator/test/nif_SUITE_data/nif_SUITE.c b/erts/emulator/test/nif_SUITE_data/nif_SUITE.c
index f10f01d14d..462848a782 100644
--- a/erts/emulator/test/nif_SUITE_data/nif_SUITE.c
+++ b/erts/emulator/test/nif_SUITE_data/nif_SUITE.c
@@ -2861,7 +2861,7 @@ static void monitor_resource_down(ErlNifEnv* env, void* obj, ErlNifPid* pid,
      * without lock order violation. */
     {
         ErlNifPid pid;
-        ErlNifPid port;
+        ErlNifPort port;
         enif_whereis_pid(env, atom_null, &pid);
         enif_whereis_port(env, atom_null, &port);
     }
